Storage/Storage/Extensions/NSManagedObjectContext+Storage.swift

```diff
@@ -175,4 +178,16 @@ extension NSManagedObjectContext: StorageType {
     private func fetchRequest<T: Object>(forType type: T.Type) -> NSFetchRequest<NSFetchRequestResult> {
         return NSFetchRequest<NSFetchRequestResult>(entityName: type.entityName)
     }
+
+    /// Helper to trigger failure in debug mode or print warning when write operations are called from the view context (main thread).
```

Comment:
Do you think using compiler flags would be better? As far as I know, this method will still be part of the production code. But compiler flags could ensure they don't show up in production.

Reply:
I moved the content of the method inside a `#if DEBUG` check to make the code run only in debug.

Reply:
Thanks! I'm gonna let @selanthiraiyan review this. 🙂